src/components/Technologies/Technologies.tsx

```diff
@@ -13,68 +18,151 @@ const searchOptions = {
+        } else {
+          return true;
+        }
+      })
+    })
     if (searchValue) {
-      const fuse = new Fuse(technologies, searchOptions);
-      technologies = fuse.search(searchValue).map(({ item }) => item);
-    }
-
-    if (selectedCategory !== "all") {
-      technologies = technologies.filter(({ fields }) => fields.category.includes(selectedCategory)) || [];
+      const fuse = new Fuse(filteredCards, searchOptions);
+      filteredCards = fuse.search(searchValue).map(({ item }) => item);
     }
+    const categoriesMap = filteredCards.reduce((acc: Map<string, any>, technology: PacksData | IntegrationsData) => {
+      let packType = technology.fields.packType;
+      if (acc.has(packType)) {
+        acc.get(packType).push(technology.fields);
+      } else {
+        acc.set(packType, [technology.fields]);
+      }
+      return acc;
+    }, new Map<string, any>());
 
-    return technologies;
-  }, [data, searchValue, selectedCategory]);
+    const sortedCategoriesMap = new Map([...categoriesMap.entries()].sort((field1: string, field2: string) => {
+      const packType1: keyof typeof packTypeNames = field1;
+      const packType2: keyof typeof packTypeNames = field2;
+      if (packTypeNames[field1[0]] > packTypeNames[field2[0]]) {
+        return 1;
+      } else if (packTypeNames[field1[0]] < packTypeNames[field2[0]]) {
+        return -1;
+      } else {
+        return 0;
+      }
+    }));
+    const categoryKeys = Array.from(sortedCategoriesMap.keys()) as string[];
+    categoryKeys.forEach((category) => {
+      let techCards: any = sortedCategoriesMap.get(category);
+      techCards.sort((a: FrontMatterData, b: FrontMatterData) => {
+        if (a.title > b.title) {
+          return 1;
+        } else if (a.title < b.title) {
+          return -1;
+        } else {
+          return 0;
+        }
+      });
+    });
+    return sortedCategoriesMap;
+  }, [data, selectedFilters, searchValue]);
 
-  const onSearch = (event: React.ChangeEvent<HTMLInputElement>) => {
-    setSearchValue(event.target.value);
+  const onSearch = (value: string) => {
+    setSearchValue(value);
+  };
+  const renderPacks = (fields: FrontMatterData[]) => {
+    return fields.map((field) => {
+      const { title, logoUrl, packType, name } = field;
+      return <TechnologyCard name={name} title={title} logoUrl={logoUrl} type={packType}></TechnologyCard>;
+    });
+  };
+  const renderPacksCategories = () => {
+    const renderedCategoryItems: React.ReactNode[] = [];
+    let categoryKeys: string[] = Array.from(filteredTechCards.keys()) as string[];
+    categoryKeys.forEach((category) => {
+      const categoryItems = filteredTechCards.get(category) as FrontMatterData[];
+      if (categoryItems.length) {
+        const obj = (<Collapse.Panel header={addPanelHeader(category)} key={category}>{renderPacks(categoryItems)}</Collapse.Panel>)
+        renderedCategoryItems.push(obj);
+      }
+    });
+    return renderedCategoryItems;
   };
```

Comment:
this is really messy.
you are creating so many new objects just to render something.
please refactor this just by using map functions.
if feels like the usage of Map might not be an inspired choice.
Forces the entire render to create arrays and then push new object inside of it. Map is faster than push. Push is mostly the slowest operation and it should be avoided if possible. I can't find the benchmark to back this up, but trust me. Push usage in render functions is not optimal